firestore/transaction.go

```diff
@@ -59,7 +62,35 @@ var ReadOnly = ro{}
 
 type ro struct{}
 
-func (ro) config(t *Transaction) { t.readOnly = true }
+func (ro) config(t *Transaction)                     { t.readOnly = true }
+func (ro) handleCommitResponse(r *pb.CommitResponse) {}
+
+// CommitResponse is a wrapper over pb.CommitResponse to avoid exposing proto directly.
+type CommitResponse struct {
+	response *pb.CommitResponse
+}
+
+// CommitTime returns the commit time from the commit response.
+func (r *CommitResponse) CommitTime() time.Time {
+	return r.response.CommitTime.AsTime()
+}
+
+// CommitResponseOption is the Transaction option to record the commit response.
+type CommitResponseOption struct {
```

Comment:
I don't understand why this can't be unexported, like `ro` and `maxAttempts`. Why does that give an error when the other two don't? If there's some sort of special case to a check, let's apply it here too.

Reply:
I don't really know, I just know that this is how it fails:

> firestore/transaction.go:96:46: exported func WithCommitResponseTo returns unexported type firestore.commitResponse, which can be annoying to use

```ro``` isn't returned from an exported function, so that wouldn't seem to apply. ```maxAttempts``` is a typedef for ```int```, not a struct, maybe that's the difference?

Reply:
@jba Looks like this error is coming is coming straight out of ```golint```. There are various discussions about this rule out there ([here](https://github.com/golang/lint/issues/465), [here](https://github.com/golang/lint/issues/210)). I don't see how it can be bypassed at the ```golint``` level.

This project (```google-cloud-go```) does seem to ignore some ```golint``` errors, i.e. [here:](https://github.com/googleapis/google-cloud-go/blob/d410907f3e52bcc64bd92e0a341777c1277a6418/.github/workflows/vet.sh#L47), but as far as I can tell adding a new exclusion there would affect the entire project. Would you like me to do that?

Reply:
Yes, add that line, except write `firestore.commitResponse`

Reply:
Done.